scheduler/plugin/wrappedplugin.go

```diff
@@ -0,0 +1,258 @@
+	opts.extenderOption = Extenders(e)
+}
+
+func WithExtendersOption(opt *Extenders) Option {
+	return extendersOption(*opt)
+}
+
+// wrappedPlugin behaves as if it is original plugin, but it records result of plugin.
+// All wrappedPlugin's name is originalPlugin name + pluginSuffix.
+type wrappedPlugin struct {
+	name                         string
+	originalFilterPlugin         framework.FilterPlugin
+	originalScorePlugin          framework.ScorePlugin
+	filterPluginExtender         FilterPluginExtender
+	scorePluginExtender          ScorePluginExtender
+	normalizeScorePluginExtender NormalizeScorePluginExtender
+	weight                       int32
+	store                        Store
+}
+
+const (
+	pluginSuffix       = "Wrapped"
+	resultBeforePrefix = "Before"
+	resultAfterPrefix  = "After"
+)
+
+func pluginName(pluginName string) string {
+	return pluginName + pluginSuffix
+}
+
+// newWrappedPlugin makes wrappedPlugin from score or/and filter plugin.
+func newWrappedPlugin(s Store, p framework.Plugin, weight int32, opts ...Option) framework.Plugin {
+	options := options{}
+	for _, o := range opts {
+		o.apply(&options)
+	}
+
+	plg := &wrappedPlugin{
+		name:   pluginName(p.Name()),
+		weight: weight,
+		store:  s,
+	}
+	if options.extenderOption.filterPluginExtender != nil {
+		plg.filterPluginExtender = options.extenderOption.filterPluginExtender
+	}
+	if options.extenderOption.scorePluginExtender != nil {
+		plg.scorePluginExtender = options.extenderOption.scorePluginExtender
+	}
+	if options.extenderOption.normalizeScorePluginExtender != nil {
+		plg.normalizeScorePluginExtender = options.extenderOption.normalizeScorePluginExtender
+	}
+
+	fp, ok := p.(framework.FilterPlugin)
+	if ok {
+		plg.originalFilterPlugin = fp
+	}
+
+	sp, ok := p.(framework.ScorePlugin)
+	if ok {
+		plg.originalScorePlugin = sp
+	}
+
+	return plg
+}
+
+func (w *wrappedPlugin) Name() string { return w.name }
+func (w *wrappedPlugin) ScoreExtensions() framework.ScoreExtensions {
+	if w.originalScorePlugin != nil && w.originalScorePlugin.ScoreExtensions() != nil {
+		return w
+	}
+	return nil
+}
+
+// NormalizeScore wraps original NormalizeScore plugin of Scheduler Framework.
+// Before and after the execution of original NormalizeScore plugin,
+// we will run arbitrary processing as functions from normalizeScorePluginExtender.
+//nolint:cyclop
+func (w *wrappedPlugin) NormalizeScore(ctx context.Context, state *framework.CycleState, pod *v1.Pod, scores framework.NodeScoreList) *framework.Status {
+	if w.originalScorePlugin == nil || w.originalScorePlugin.ScoreExtensions() == nil {
+		// return nil not to affect scoring
+		return nil
+	}
+
+	if w.normalizeScorePluginExtender != nil {
+		s := w.normalizeScorePluginExtender.BeforeNormalizeScore(ctx, state, pod, scores)
+		for _, score := range scores {
+			w.store.AddNormalizedScoreResult(pod.Namespace, pod.Name, score.Name, resultBeforePrefix+w.originalScorePlugin.Name(), score.Score)
+		}
```

Comment:
L154-L156 are not needed.

Reply:
Thanks.
Indeed, there was no description of those storing processes. It was my own mistake.
I fix these.